flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/HiveCatalog.java

```diff
@@ -374,8 +375,30 @@ public void createTable(ObjectPath tablePath, CatalogBaseTable table, boolean ig
 
 		Table hiveTable = instantiateHiveTable(tablePath, table);
 
+		UniqueConstraint pkConstraint = null;
+		List<String> notNullCols = new ArrayList<>();
+		boolean isGeneric = isGenericForCreate(table.getOptions());
+		if (!isGeneric) {
+			pkConstraint = table.getSchema().getPrimaryKey().orElse(null);
+			for (int i = 0; i < table.getSchema().getFieldDataTypes().length; i++) {
+				if (!table.getSchema().getFieldDataTypes()[i].getLogicalType().isNullable()) {
+					notNullCols.add(table.getSchema().getFieldNames()[i]);
+				}
+			}
+		}
+
 		try {
-			client.createTable(hiveTable);
+			if (pkConstraint != null || !notNullCols.isEmpty()) {
+				// for now we just create constraints that are DISABLE, NOVALIDATE, RELY
+				Byte[] pkTraits = new Byte[pkConstraint == null ? 0 : pkConstraint.getColumns().size()];
+				Arrays.fill(pkTraits, HiveTableUtil.relyConstraint((byte) 0));
+				Byte[] nnTraits = new Byte[notNullCols.size()];
+				Arrays.fill(nnTraits, HiveTableUtil.relyConstraint((byte) 0));
+				client.createTableWithConstraints(hiveTable, hiveConf,
```

Comment:
Can client do more things to include upper couple of lines?

Reply:
The client is meant to be just a wrapper of `IMetaStoreClient`. So I think it's more natural to keep such logic in `HiveCatalog`. We can add util method if that logic has to be repeated somewhere else.